third_party/bigframes_vendored/pandas/io/gbq.py

```diff
@@ -28,16 +29,25 @@ def read_gbq(
         """Loads a DataFrame from BigQuery.
 
         BigQuery tables are an unordered, unindexed data source. To add support
-        pandas-compatibility, the following indexing options are supported:
-
-        * (Default behavior) Add an arbitrary sequential index and ordering
-          using an an analytic windowed operation that prevents filtering
-          push down.
+        pandas-compatibility, the following indexing options are supported via
+        the ``index_col`` parameter:
+
+        * (Empty iterable, default) A default index. **Behavior may change.**
+          Explicitly set ``index_col`` if your application makes use of
+          specific index values.
+
+          If a table has primary key(s), those are used as the index,
+          otherwise a sequential index is generated.
+        * (:attr:`bigframes.enums.DefaultIndexKind.SEQUENTIAL_INT64`) Add an
+          arbitrary sequential index and ordering. **Warning** This uses an
+          analytic windowed operation that prevents filtering push down. Avoid
+          using on large clustered or partitioned tables.
         * (Recommended) Set the ``index_col`` argument to one or more columns.
           Unique values for the row labels are recommended. Duplicate labels
           are possible, but note that joins on a non-unique index can duplicate
-          rows and operations like ``cumsum()`` that window across a non-unique
-          index can have some non-deternimism.
+          rows via pandas-like outer join behavior. Operations like
+          ``cumsum()`` that window across a non-unique index can have some
+          unpredictability due to ambiguous ordering.
```

Comment:
The part about non-determinism I don't think is correct? My understanding is if the index is non-unique, we fall back to hidden hashes to ensure total ordering.

Reply:
Dropped this.